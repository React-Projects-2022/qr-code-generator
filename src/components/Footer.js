export const Footer = () => (
  <div class="footer">
    <p>QR Code Generator - Anartz Mugika Ledo - { new Date().getFullYear()}</p>
  </div>
);
