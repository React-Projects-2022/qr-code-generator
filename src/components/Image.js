export const Image = ({ qrCode, text }) => (
  <img src={qrCode} alt={"QR Code Image with ".concat(text)} />
);
