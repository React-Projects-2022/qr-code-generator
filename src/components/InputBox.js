export const InputBox = ({
  size,
  temp,
  changeSize,
  changeBgColor,
  asignInputTempText,
  asignFindText,
}) => {
  const handleClick = () => asignFindText(temp);
  return (
    <div className="input-box">
      <div className="gen">
        <input
          type="text"
          autoFocus={true}
          onChange={(e) => {
            asignInputTempText(e.target.value);
          }}
          placeholder="Enter text to encode"
        />
        <button className="button" onClick={handleClick}>
          Generate
        </button>
      </div>
      <div className="extra">
        <h5>Background Color:</h5>
        <input
          type="color"
          onChange={(e) => {
            changeBgColor(e.target.value.substring(1));
          }}
        />
        <h5>Dimension:</h5>
        <input
          type="range"
          min="200"
          max="600"
          value={size}
          onChange={(e) => {
            changeSize(e.target.value);
          }}
        />
      </div>
    </div>
  );
};
