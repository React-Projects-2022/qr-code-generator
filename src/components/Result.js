import { Image } from "./Image";
import { downloadImage } from "./../helpers/download-image";
export const Result = ({ qrCode, text }) => (
  <div className="output-box">
    <Image qrCode={qrCode} text={text} />
    <button onClick={() => downloadImage(qrCode, text)} type="button">
      Download
    </button>
  </div>
);
