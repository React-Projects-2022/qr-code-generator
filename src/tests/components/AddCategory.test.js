import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { AddCategory } from "../../components/AddCategory";
describe("Pruebas en AddCategory", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);;
  beforeEach(() => {
    // Para limpiar los mocks de Jest
    jest.clearAllMocks();
    // Inicializamos el componente
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("Debe de mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    // Importante añadir el valor de target: value
    input.simulate("change", { target: { value: "Hola mundo" } });
  });

  test("NO debe de postear la información 'onSubmit'", () => {
    const form = wrapper.find("form");
    form.simulate("submit", { preventDefault(){}});

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de postear la información 'onSubmit' y limpiar input", () => {
    
    // 1.- Simular input change  - Importante añadir el valor de target: value
    wrapper.find("input").simulate("change", { target: { value: "Hola mundo" } });

    // 2.- Ejecutar submit
    wrapper.find("form").simulate("submit", { preventDefault(){}});

    // 3.- Comprobar que setCategories ha sido llamado
    expect(setCategories).toHaveBeenCalled(); // Ha sido llamada una vez por lo menos
    expect(setCategories).toHaveBeenCalledTimes(1); // Veces llamadas
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); // Llamada de otra función

    // 4.- Comprobar que se ha limpiado correctamente
    expect(wrapper.find("input").prop('value')).toBe('');
  });
});
