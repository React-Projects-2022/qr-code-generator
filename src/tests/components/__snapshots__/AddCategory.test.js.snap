// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe de mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
